spec/globus_client_spec.rb

```diff
@@ -98,16 +98,47 @@
   end
 
   describe "#user_exists?" do
-    let(:fake_identity) { instance_double(described_class::Identity, exists?: nil) }
+    context "when request is successful" do
+      let(:fake_identity) { instance_double(described_class::Identity, exists?: nil) }
 
-    before do
-      allow(described_class::Identity).to receive(:new).and_return(fake_identity)
-      allow(fake_identity).to receive(:exists?)
+      before do
+        allow(described_class::Identity).to receive(:new).and_return(fake_identity)
+        allow(fake_identity).to receive(:exists?)
+      end
+
+      it "invokes Identity#exists?" do
+        client.user_exists?(sunetid: "bogus")
+        expect(fake_identity).to have_received(:exists?).once
+      end
     end
 
-    it "invokes Identity#exists?" do
-      client.user_exists?(sunetid: "bogus")
-      expect(fake_identity).to have_received(:exists?).once
+    context "when token is expired" do
+      let(:fake_identity) { instance_double(described_class::Identity, exists?: nil) }
```

Comment:
I wasn't sure how to best test this new TokenWrapper. The various scenarios are tested here for the `Identity#user_exists?` instance method. Should I add this kind of testing to all methods (or any another method in particular)?

Reply:
Testing `TokenWrapper` in the context of `Identity#user_exists?` makes sense to me since that's the area of the code that led us to create it. I don't think testing other methods with the token wrapper will get us much given `TokenWrapper` is already fully covered. We could consider adding a specific `token_wrapper_spec` but, same thing, I think this test covers it. Maybe it'd be sufficient to toss a comment above the block that tests the token wrapper to call out that we're using it as test coverage for the wrapper with a brief rationale (which you and I may have pretty much already written in our comments here). What do you think?